src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -399,13 +400,31 @@ ErrorIfQueryNotSupported(Query *queryTree)
 		errorHint = filterHint;
 	}
 
-	if (queryTree->setOperations)
+	if (queryTree->setOperations && !subqueryPushdown)
```

Comment:
I'm thinking that we should have this check in `ErrorIfCannotPushdownSubquery()` function. What do you think?